__tests__/app.test.js

```diff
@@ -147,3 +147,63 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe.only("POST /api/articles/:article_id/comments", () => {
+  it("201: responds with a newly created comment object", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Salted pistachio is the best ice cream flavour",
+    };
+    return request(app)
+      .post("/api/articles/6/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toMatchObject({
+          comment_id: 19,
+          votes: 0,
+          created_at: expect.any(String),
+          author: "butter_bridge",
+          body: "Salted pistachio is the best ice cream flavour",
+          article_id: 6,
+        });
+      });
+  });
+  it("404: responds with an error message if an article with given id does not exist", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Salted pistachio is the best ice cream flavour",
+    };
+    return request(app)
+      .post("/api/articles/231/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Article not found");
+      });
+  }, 20000);
+  it("400: responds with an error message for invalid article id", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Salted pistachio is the best ice cream flavour",
+    };
+    return request(app)
+      .post("/api/articles/eleven/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  it("400: responds with an error message for incomplete data in a request", () => {
+    const newComment = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
```

Comment:
Another 404 test for a non-existent user is worth adding here.

Reply:
Author in the comments table is a foreign key which relates to username in the users table. A psql error will be thrown if this is violated. It is not necessary to separately check if the user exists.